avr-hal-generic/src/serial.rs

```diff
@@ -52,10 +79,8 @@ macro_rules! impl_usart {
                 usart
             }
 
-            fn initialize(&mut self, baud: u32) {
-                // Value for baudrate register must be calculated based on clock frequency.
-                let brr = CLOCK::FREQ / (16 * baud) - 1;
-                self.p.[<ubrr $n>].write(|w| unsafe { w.bits(brr as u16) });
+            fn initialize(&mut self, baud: Baudrate<CLOCK>) {
+                self.p.[<ubrr $n>].write(|w| unsafe { w.bits(baud.ubrr) });
```

Comment:
The `u2x` bit needs to be written as well now.  It's in `UCSRnA`, so replace the `.reset()` with something similar to:

```rust
self.p.[<ucsr $n a>].write(|w| w.[<u2x $n>].bit(baud.u2x));
```

Reply:
done